inject/src/test/groovy/io/micronaut/context/env/PropertySourcePropertyResolverSpec.groovy

```diff
@@ -641,4 +643,70 @@ class PropertySourcePropertyResolverSpec extends Specification {
         then:
         resolver.containsProperty("extra.listval")
     }
+
+    void "test expression resolver"() {
+        given:
+            Map<String, Object> parameters = [foo: "bar"]
+            PropertyResolver propertyResolver = new MapPropertyResolver(parameters)
+            DefaultPropertyPlaceholderResolver propertyPlaceholderResolver = new DefaultPropertyPlaceholderResolver(propertyResolver, ConversionService.SHARED);
+            propertyPlaceholderResolver.@expressionResolvers = [new PropertyExpressionResolver() {
+                @Override
+                Optional resolve(PropertyResolver pr, ConversionService conversionService, String expression, Class requiredType) {
+                    assert requiredType == String.class
+                    assert conversionService
+                    if ("foobar" == expression) {
+                        return Optional.of("ABC")
+                    }
+                    if ("xyz" == expression) {
+                        return Optional.of("123")
+                    }
+                    if ("external" == expression) {
+                        return pr.get("foo", requiredType)
+                    }
+                    return Optional.empty()
+                }
+            }]
+
+        expect:
+            def resolved = propertyPlaceholderResolver.resolvePlaceholders(template)
+            if (result) {
+                resolved.get() == result
```

Comment:
Please, add `assert`. Without it this line is not testing anything.

Reply:
I added it here: https://github.com/micronaut-projects/micronaut-core/pull/7422/commits/8bd17737105bec0903902c2828594bed9f8bd44a